.github/workflows/buildtest.yml

```diff
@@ -0,0 +1,147 @@
+---
+# Tools that can save round-trips to github and a lot of time:
+#
+# yamllint -f parsable pull_request.yml
+# pip3 install ruamel.yaml.cmd
+# yaml merge-expand pull_request.yml exp.yml &&
+#    diff -w -u pull_request.yml exp.yml
+#
+# github.com also has a powerful web editor that can be used without
+# committing.
+
+# This is the name of this workflow and should technically be called
+# something like "Main Workflow" but the place where most people see
+# this name is the Checks window next to other, non-github checks.
+name: build test
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+    branches:
+      - topic/sof-dev
+  pull_request:
+    branches:
+      - topic/sof-dev
+
+jobs:
+  sparse-check:
+    name: "Sparse check"
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Fetch kconfig
+        uses: actions/checkout@v2
+        with:
+          repository: 'thesofproject/kconfig'
+          path: 'kconfig'
+
+      - name: Install libelf
+        run: sudo apt install -y libelf-dev
+
+      - name: Install sparse
+        run: |
+          git clone --depth 1 --branch v0.6.2 git://git.kernel.org/pub/scm/devel/sparse/sparse.git
+          BINDIR=$(pwd)/bin make -C sparse install
+          echo "$(pwd)/bin" >> $GITHUB_PATH
+
+      - name: build start
+        run: |
+          export ARCH=x86_64
+          bash kconfig/kconfig-sof-default.sh
+          make modules_prepare
+          make M=sound/soc/sof C=2
+
+  gcc-sof-x86_64:
+    name: "GCC build X86_64"
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Fetch kconfig
+        uses: actions/checkout@v2
+        with:
+          repository: 'thesofproject/kconfig'
+          path: 'kconfig'
+
+      - name: Install libelf
+        run: sudo apt install -y libelf-dev
+
+      - name: build start
+        run: |
+          export ARCH=x86_64 KCFLAGS="-Wall -Werror"
+          export MAKEFLAGS=j"$(nproc)"
+          bash kconfig/kconfig-sof-default.sh
+          make sound/ W=1
+          make drivers/soundwire/ W=1
```

Comment:
Could we split the W=1 compilation in a separate test? that way we can know when W=1 fails but regular compilation is fine.

Reply:
sure, that would help us to check warning and errors.